plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse_subscription.go

```diff
@@ -0,0 +1,115 @@
+
+var queueBehaviours = []string{"", "DiscardOldest", "DiscardNewest"}
+var valueChanges = []string{"", "Status", "StatusValue", "StatusValueTimestamp"}
+
+type Subscription struct {
+	index             int
+	Nodes             []string        `toml:"nodes"`
+	Measurement       string          `toml:"measurement"`
+	PublishInterval   config.Duration `toml:"publish_interval"`
+	KeepaliveInterval config.Duration `toml:"keep_alive_interval"`
+	ErrorInterval     config.Duration `toml:"error_interval"`
+	SamplingInterval  config.Duration `toml:"sampling_interval"`
+	QueueSize         uint            `toml:"queue_size"`
+	QueueBehaviour    string          `toml:"queue_behaviour"`
+	DeadBandValue     float64         `toml:"dead_band_value"`
+	ValueChange       string          `toml:"value_change"`
+}
+
+type Rule struct {
+	RuleType string      `json:"rule_type"`
+	Rule     interface{} `json:"rule"`
+}
+
+type Sampling struct {
+	SamplingInterval uint64 `json:"samplingInterval"`
+}
+
+type Queueing struct {
+	QueueSize uint   `json:"queueSize"`
+	Behaviour string `json:"behaviour"`
+}
+
+type DataChangeFilter struct {
+	DeadBandValue float64 `json:"deadBandValue"`
+}
+
+type ChangeEvents struct {
+	ValueChange      string `json:"valueChange"`
+	BrowselistChange bool   `json:"browselistChange"`
+	MetadataChange   bool   `json:"metadataChange"`
+}
+
+type SubscriptionProperties struct {
+	KeepaliveInterval int64  `json:"keepaliveInterval"`
+	Rules             []Rule `json:"rules"`
+	ID                string `json:"id"`
+	PublishInterval   int64  `json:"publishInterval"`
+	ErrorInterval     int64  `json:"errorInterval"`
+}
+
+// subscriptionRequestBody can be used to to create a sse subscription at the ctrlX Data Layer.
+type SubscriptionRequestBody struct {
+	Properties SubscriptionProperties `json:"properties"`
+	Nodes      []string               `json:"nodes"`
+}
+
+// applyDefaultSettings applies the default settings if they are not configured in the config file.
+func (s *Subscription) applyDefaultSettings() {
+	if s.Measurement == "" {
+		s.Measurement = defaultMeasurementName
+	}
+	if s.PublishInterval == 0 {
+		s.PublishInterval = defaultPublishInterval
+	}
+	if s.KeepaliveInterval == 0 {
+		s.KeepaliveInterval = defaultKeepaliveInterval
+	}
+	if s.ErrorInterval == 0 {
+		s.ErrorInterval = defaultErrorInterval
+	}
+	if s.SamplingInterval == 0 {
+		s.SamplingInterval = defaultSamplingInterval
+	}
+	if s.QueueSize == 0 {
+		s.QueueSize = defaultQueueSize
+	}
+	if s.QueueBehaviour == "" {
+		s.QueueBehaviour = defaultQueueBehaviour
+	}
+	if s.ValueChange == "" {
+		s.ValueChange = defaultValueChange
+	}
+}
+
+// createRequestBody builds the request body for the sse subscription, based on the subscription configuration.
+// The request body can be send to the server to create a new subscription.
+func (s *Subscription) createRequestBody() SubscriptionRequestBody {
+	s.applyDefaultSettings()
```

Comment:
It would be good if the main `Init()` would call this function (`s.applyDefaultSettings()`) only once.

Reply:
Done.